src/main/scala/fs2/aws/kinesis/KinesisSettings.scala

```diff
@@ -0,0 +1,47 @@
+package fs2
+package aws
+package kinesis
+
+import scala.concurrent.duration._
+
+/** Settings for configuring the Kinesis consumer stream
+  *
+  *  @param bufferSize size of the internal buffer used when reading messages from Kinesis
+  */
+class KinesisStreamSettings private (val bufferSize: Int)
+
+/** Settings for configuring the Kinesis checkpointer pipe
+  *
+  *  @param maxBatchSize the maximum number of records to aggregate before checkpointing the cluster of records. Passing 1 means checkpoint on every record
+  *  @param maxBatchWait the maximum amount of time to wait before checkpointing the cluster of records
+  */
+class KinesisCheckpointSettings private (
+    val maxBatchSize: Int,
+    val maxBatchWait: FiniteDuration
+)
+
+object KinesisStreamSettings {
+  val defaultInstance: KinesisStreamSettings = new KinesisStreamSettings(10)
+
+  def apply(bufferSize: Int): Either[Throwable, KinesisStreamSettings] =
+    if (bufferSize < 1)
+      Left(new IllegalArgumentException("Buffer size must be greater than 0"))
+    else
+      Right(new KinesisStreamSettings(bufferSize))
+}
+
+object KinesisCheckpointSettings {
+  val defaultInstance = new KinesisCheckpointSettings(1000, 10.seconds)
+
+  def apply(maxBatchSize: Int,
+            maxBatchWait: FiniteDuration): Either[Throwable, KinesisCheckpointSettings] =
```

Comment:
There are multiple types of KinesisSettings Exception so I think you can create a sealed trait KinesisSettingsException and then case object / case classes that represent the specific errors

Reply:
neat, sounds good